src/main/java/com/fincons/service/employeeService/impl/DepartmentServiceImpl.java

```diff
@@ -24,128 +23,109 @@ public class DepartmentServiceImpl implements DepartmentService {
-
-        Department department = modelMapperProject.mapToEntity(departmentDTO);
+        checkForDuplicateDepartment(department.getDepartmentCode(), department.getName());
 
         departmentRepository.save(department);
 
         return department;
     }
-
     @Override
-    public Department updateDepartmentByCode(String departmentCode, DepartmentDTO departmentDTO) {
-
-        ValidateSingleField.validateSingleField(departmentCode);
-        //Condition for not have null attributes
-        validateDepartmentFields(departmentDTO);
+    public Department updateDepartmentByCode(String departmentCode, Department department) {
 
         List<Department> departments = departmentRepository.findAll();
 
-        //Check if the specified ID exists
-        Department department = validateDepartmentByCode(departmentCode);
+        //Check if the specified CODE exists
+        Department departmentExisting = validateDepartmentByCode(departmentCode);
 
         List<Department> departmentsWithoutDepartmentCodeChosed = new ArrayList<>();
 
-        for (Department d : departments ) {
-            if(!Objects.equals(d.getDepartmentCode(), departmentCode)){
+        for (Department d : departments) {
+            if (!Objects.equals(d.getDepartmentCode(), departmentCode)) {
                 departmentsWithoutDepartmentCodeChosed.add(d);
             }
         }
 
         //Set the new instance department
-        department.setDepartmentCode(departmentDTO.getDepartmentCode());
-        department.setName(departmentDTO.getName());
-        department.setAddress(departmentDTO.getAddress());
-        department.setCity(departmentDTO.getCity());
-
-        if(departmentsWithoutDepartmentCodeChosed.isEmpty()){
-            departmentRepository.save(department);
-        }
-        else {
-            for (Department d : departmentsWithoutDepartmentCodeChosed ) {
-                if(d.getDepartmentCode().equals(department.getDepartmentCode())){
-                    throw new DuplicateException("This code: " + departmentDTO.getDepartmentCode() + " is already taken" );
-                } else if (d.getName().equals(department.getName())) {
-                    throw new DuplicateException("This name " + departmentDTO.getName() + " is already taken");
+        departmentExisting.setDepartmentCode(department.getDepartmentCode());
+        departmentExisting.setName(department.getName());
+        departmentExisting.setAddress(department.getAddress());
+        departmentExisting.setCity(department.getCity());
+
+        if (departmentsWithoutDepartmentCodeChosed.isEmpty()) {
+            departmentRepository.save(departmentExisting);
+        } else {
+            for (Department d : departmentsWithoutDepartmentCodeChosed) {
+                if (d.getDepartmentCode().equals(departmentExisting.getDepartmentCode())) {
+                    throw new DuplicateException("This code: " + department.getDepartmentCode() + " is already taken");
```

Comment:
Cerca di dare piu informazioni iniziali riguardo l'operazione nell'eccezioni esempio: 
"Errore durante la creazione del Department: ........"
"Errore durante la cancellazione Department: ........."
In questo modo sappiamo in quale operazione si è verificato l'errore e il perché. Logicamente in inglese